src/coreclr/src/vm/dllimportcallback.cpp

```diff
@@ -614,44 +615,124 @@ VOID UMEntryThunk::CompileUMThunkWorker(UMThunkStubInfo *pInfo,
     pcpusl->X86EmitNearJump(pEnableRejoin);
 }
 
+namespace
+{
+    // Templated function to compute if a char string begins with a constant string.
+    template<size_t S2LEN>
+    bool BeginsWith(ULONG s1Len, const char* s1, const char (&s2)[S2LEN])
+    {
+        WRAPPER_NO_CONTRACT;
+
+        ULONG s2Len = (ULONG)S2LEN - 1; // Remove null
+        ULONG minLen = s1Len < s2Len ? s1Len : s2Len;
```

Comment:
Should this rather return false when the actual length is less then desired length?

Reply:
As in a preemptive check? i.e. `s1` is less than `s2` therefore just return `false`?

Reply:
Also for correctness. E.g. if the name is going to be just `S`, I believe the current implementation is going to return `true`.

Reply:
Yes, I agree that is possible. In this context I had assumed the FQN would always be much longer, but that isn't captured in this function nor upon further reflection am I confident that is always true.